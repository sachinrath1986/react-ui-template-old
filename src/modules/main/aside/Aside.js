import Brand from "../brand/Brand"
import AsideMenu from "../aside-menu/AsideMenu";

const Aside = () => {
    return (
        <div className="aside d-flex flex-column flex-row-auto">
            <Brand />
            <AsideMenu />
        </div>
    );
}

export default Aside;