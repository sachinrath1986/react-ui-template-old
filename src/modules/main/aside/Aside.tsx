import { useSelector } from "react-redux";
import { useHistory } from "react-router-dom";
import SVG from "react-inlinesvg";
import Brand from "../brand/Brand";
import AsideMenu from "../aside-menu/AsideMenu";
import logoutSVG from "../../../theme/assets/media/svg/Home/Door-open.svg";

const Aside = () => {
  const history = useHistory();

  const isSideBarCollapsed = useSelector(
    (state: any) => state.app.isSidebarCollapsed
  );

  const logout = () => {
    history.push("/auth/login");
  };
  return (
    <div
      className={
        `aside d-flex flex-column flex-row-auto ` +
        (isSideBarCollapsed ? "isCollapsed" : "")
      }
    >
      <Brand />
      <AsideMenu />
      <div className="logout-ct d-flex flex-center" onClick={() => logout()}>
        <SVG src={logoutSVG} />
        <span>Logout</span>
      </div>
    </div>
  );
};

export default Aside;
