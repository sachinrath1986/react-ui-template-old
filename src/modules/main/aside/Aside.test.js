import { render } from "@testing-library/react";
import * as ReactRedux from "react-redux";
import Aside from "./Aside";

jest.mock("../brand/Brand", () => () => <div>Mock Brand component</div>);
jest.mock("../aside-menu/AsideMenu", () => () => (
  <div>Mock AsideMenu component</div>
));

describe("Main", () => {
  const useSelectorSpy = jest.spyOn(ReactRedux, "useSelector");

  it("renders snapshots when isSidebarCollapsed=true", () => {
    useSelectorSpy.mockReturnValue({
      app: {
        isSidebarCollapsed: true,
      },
    });
    const { container } = render(<Aside />);
    expect(container).toMatchSnapshot();
  });

  it("renders snapshots when isSidebarCollapsed=false", () => {
    useSelectorSpy.mockReturnValue({
      app: {
        isSidebarCollapsed: false,
      },
    });
    const { container } = render(<Aside />);
    expect(container).toMatchSnapshot();
  });
});
