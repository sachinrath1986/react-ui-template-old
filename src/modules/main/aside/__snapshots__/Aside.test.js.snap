// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders snapshots when isSidebarCollapsed=false 1`] = `
<div>
  <div
    class="aside d-flex flex-column flex-row-auto isCollapsed"
  >
    <div>
      Mock Brand component
    </div>
    <div>
      Mock AsideMenu component
    </div>
    <div
      class="logout-ct d-flex flex-center"
    >
      <span>
        Logout
      </span>
    </div>
  </div>
</div>
`;

exports[`Main renders snapshots when isSidebarCollapsed=true 1`] = `
<div>
  <div
    class="aside d-flex flex-column flex-row-auto isCollapsed"
  >
    <div>
      Mock Brand component
    </div>
    <div>
      Mock AsideMenu component
    </div>
    <div
      class="logout-ct d-flex flex-center"
    >
      <span>
        Logout
      </span>
    </div>
  </div>
</div>
`;
